### ISSUE TITLE
Composer project type incorrectly detected (library or project)

### ISSUE THREAD

**Issue by bburnichon** [renovatebot/renovate#13795]
### What would you like Renovate to be able to do?

I have composer project type defined as "bundle", which actually means it is a symfony bundle and should be considered as a library from dependencies point of view.

I did not find a way to indicate renovate that a project is a library or a project

Sadly, renovate by default pin its dependencies which does not work with composer.
It is fine to pin devdependencies though. I wanted to keep `rangeStrategy` to `auto` and not resort to `rangeStrategy: replace` for all dependency types (`dependencies`, `devdependencies`, `peerdependencies`)

### If you have any ideas on how this should be implemented, please tell us here.

For composer manager, project types can be anything you want. Maybe add a parameter to indicate which types should be considered as libraries by renovate.

for example: `projectDependencyType: library` for single project override
or for packageManager detected as composer, a list of types that should be detected as projects or library.
I think that renovate should default to `library`, and not the current way where only applies library type when `"type": "library"` in composer.json file.
```
{
   "packageManager": "composer",
   "libraryTypes": ["library", "bundle", "composer-plugin", "wp-plugin", ...],
   "projectTypes": ["project", "application", "cli", ...],
   "defaultDependencyManagementType": "library"
}
```

### Is this a feature you are interested in implementing yourself?

No

**Comment by rarkins** [comment]
https://github.com/renovatebot/renovate/blob/63f8371b93c74cf8a61e76de1b281e10b8ac57b9/lib/manager/composer/extract.ts#L178

https://github.com/renovatebot/renovate/blob/63f8371b93c74cf8a61e76de1b281e10b8ac57b9/lib/manager/composer/range.ts#L31-L38

**Comment by viceice** [comment]
You can simply exclude the pin dep preset.

https://docs.renovatebot.com/configuration-options/#ignorepresets
 https://docs.renovatebot.com/presets-default/#pindependencies

**Comment by bburnichon** [comment]
Nice, did not know that, will definitively try to ignore pindependencies preset.

Also, feature could be done via parameterizing given list for isApp constant in given snippet
https://github.com/renovatebot/renovate/blob/63f8371b93c74cf8a61e76de1b281e10b8ac57b9/lib/manager/composer/range.ts#L31-L38

**Comment by bburnichon** [comment]
> You can simply exclude the pin dep preset.
> 
> https://docs.renovatebot.com/configuration-options/#ignorepresets https://docs.renovatebot.com/presets-default/#pindependencies

I tried to exclude the pin dep preset, I also tried several configurations and I did not managed to have the expected pinning of dev Dependencies. My last try is:
```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base"
  ],
  "labels": ["dependencies"],
  "separateMultipleMajor": true,
  "major": {
    "dependencyDashboardApproval": true
  },
  "packageRules": [
    {
      "matchPackagePatterns": [
        "*"
      ],
      "rangeStrategy": "replace"
    },
    {
      "matchDepTypes": [
        "devDependencies"
      ],
      "rangeStrategy": "pin"
    },
    {
      "matchDepTypes": [
        "peerDependencies"
      ],
      "rangeStrategy": "widen"
    }
  ]
}
```

which is trying to apply `:pinOnlyDevDependencies` manually (I also tried to simply add `":pinOnlyDevDependencies"` to extends list)

**Comment by mklenk** [comment]
We have the exact same problem with a bunch of our symfony bundles, which are incorrectly identified as type "app".
Though we have defined the type as "symfony-bundle" as per official documentation: https://symfony.com/doc/current/bundles/best_practices.html#composer-metadata

I have also tried to pin the dev dependencies manually with a nearly identical preset as @bburnichon :

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base",
    ":prHourlyLimitNone",
    ":labels(dependencies, bundle)"
  ],
  "packageRules": [
    {
      "matchPackagePatterns": [
        "*"
      ],
      "rangeStrategy": "bump"
    },
    {
      "matchDepTypes": [
        "devDependencies"
      ],
      "rangeStrategy": "pin"
    },
    {
      "matchDepTypes": [
        "peerDependencies"
      ],
      "rangeStrategy": "widen"
    }
  ],
  "reviewersFromCodeOwners": true
}
```
which is resulting in a MR like this:

![image](https://user-images.githubusercontent.com/5739238/153187184-ac903103-18ea-4562-96b7-257572eafdd5.png)

This looks to me like renovate is just bumping the dev dependency and completely ignoring the `matchDepTypes: devDependencies` part.

So basically we have two problems:

1) composer `type: symfony-bundle` is not detected as library (then we could just leave the `rangeStrategy` as `auto`)
2) `rangeStrategy: pin` is not working with this specific preset (which is just a copy of [`:pinOnlyDevDependencies`](https://docs.renovatebot.com/presets-default/#pinonlydevdependencies))

The only thing I can think of is that a composer package/bundle doesn't have composer.lock file since a library has to be able to work with any version specified in the composer.json. Maybe that's why renovate won't pin the dev dependencies?
